### ISSUE TITLE
`UnTeX` option

### ISSUE THREAD

**Issue by pihart** [cjquines/qboard#187]
When an image that was inserted from qboard's tex system is selected, users should be able to run an action that recovers the original tex source. This should also be in the image's context menu (see also #28). This should work even if the file is saved and reopened.

Solution:
* When adding an image from the tex action, tag it with a data field containing the original source.
  * We can do this by adding a data tag to the generated svg before importing; it should be preserved when serialized. Note that this would require us to parse the svg+xml as xml, which for most svgs is a very small but nonzero amount of work. The next option doesn't require this.
  * Alternatively, we can tell fabric itself what the source was by giving it to a field in the `fabric.Image` element. Figure out how to make sure it gets serialized and reinserted.
    * Make sure to make a custom type `TeXImage` or something so you don't have a bunch of floating `any`s and invalid lookups. The operation that checks whether the image came from tex should be a type guard for this type.
* The action should check that there is a selection and that it is valid by whatever spec we define. Then extract the tex and return it according to whatever spec.
  * It can take a flag so that if we know that the selection is already an image or image from tex we don't need to check again.
* For any image selection from tex (or whatever spec), there should be a context menu item that invokes the action.

Spec decision:
* What to do when the selection contains more than one valid image file
  * You could just have it stop checking when it realizes there is more than one item in the selection. This seems like an unnecessary restriction
  * You could have it output the tex values of *all* valid images from tex in some sensible way
* How to return the value?
  * console
  * alert + console
  * clipboard copy + console
    * What would this copy to the clipboard if there is more than one admissible image file?

---

This is again not a security concern because we never render arbitrary tex (except obviously when the immediate user requests that we do so with the tex action).

However, we may want to be mindful that someone may make a fake JSON file where they set the tex data to a wrong value. I think this is fine; the user should always validate their own TeX before running it, but we might want to give a warning that the returned tex is not necessarily correct.

Keep in mind that if you actually do something about this, you also want to cleanse every other SVG added to the board. Also, it is not a valid solution to check the validity of the tex field by running it and comparing the outputs, because the whole security concern is that you don't want to run it (we don't trust mathjax enough to say that it doesn't have an XSS or blocking operation vulnerability). I don't think this is something to worry about, though (just like it's not Google Docs's job to ensure that what someone writes in a comment is truthful).

**Comment by cjquines** [comment]
see #188 for my feelings on this

**Comment by cjquines** [comment]
in particular i do not see any use case at all for this

**Comment by pihart** [comment]
> see #188 for my feelings on this

I disagree; this feature is just reporting information that we *already have*.

> in particular i do not see any use case at all for this. notably, this is _not_ a feature requested in _anything_ because _there is no use case_ and i will insist there is no use case

I have wanted to edit formulas/text I've added to qboard many times. This let's us do precisely that.

**Comment by cjquines** [comment]
> I have wanted to edit formulas/text I've added to qboard many times

okay, sure, that's a use case. and yet, i don't think there's any good ui for this. nothing in qboard currently allows you to edit it once it's placed. change the color of something you drew? idk. maybe. but still

**Comment by pihart** [comment]
> i don't think there's any good ui for this.

Just an action is sufficient. The bulk of the work is then #189, which is already solved by #190.

> nothing in qboard currently allows you to edit it once it's placed. change the color of something you drew? idk. maybe. but still

This is precisely #28. If we want to add anything beyond an action (we might not, and I'm personally fine that way) then we'll just piggy-back on that idea, with #28 ideally but not necessarily resolved first.

**Comment by cjquines** [comment]
i think this is okay now, actually, but i think this is blocked by #28 first

**Comment by pihart** [comment]
> i think this is blocked by #28 first

Changed this issue to be just the action; context menu part has been moved to #203
